bundles/org.eclipse.jface/src/org/eclipse/jface/viewers/AbstractTableViewer.java

```diff
@@ -885,6 +958,27 @@ protected void setSelectionToWidget(@SuppressWarnings("rawtypes") List list, boo
 					System.arraycopy(indices, 0, indices = new int[count], 0, count);
 				}
 
+				// item to select may be hidden inside expandable node.
+				if (indices.length < list.size() && getLastElement() instanceof IExpandableNode expNode) {
+					Object[] remEles = expNode.getRemainingElements();
+					boolean breakOuter = false;
+					for (Object searchItem : list) {
+						for (Object remEle : remEles) {
+							if (equals(remEle, searchItem)) {
+								int[] placeHolder = new int[indices.length + 1];
+								System.arraycopy(indices, 0, placeHolder, 0, indices.length);
+								placeHolder[placeHolder.length - 1] = items.length - 1;
```

Comment:
`= items.length - 1` not sure I understood that. Why do we **always** set index to the last element in the `items`? This for loop doesn't look OK.

Reply:
we have entered here after knowing last item is ExpandableNode. And now we need to set selection to that because we found at least one item in remaining elements of ExpandableNode.